src/main/java/graphql/execution/Execution.java

```diff
@@ -100,11 +100,11 @@ private ExecutionResult executeOperation(ExecutionContext executionContext, Obje
         ExecutionResult result;
         try {
             if (operation == OperationDefinition.Operation.MUTATION) {
-                result = mutationStrategy.execute(executionContext, parameters);
+                result = mutationStrategy.execute(executionContext, parameters).toCompletableFuture().join();
             } else if (operation == SUBSCRIPTION) {
-                result = subscriptionStrategy.execute(executionContext, parameters);
+                result = subscriptionStrategy.execute(executionContext, parameters).toCompletableFuture().join();
             } else {
-                result = queryStrategy.execute(executionContext, parameters);
+                result = queryStrategy.execute(executionContext, parameters).toCompletableFuture().join();
             }
```

Comment:
Perhaps they will evolve the `CompletionStage` interface to make it useful. `toCompletableFuture().get()` is not horrible imo.